source/MaterialXGenOgsFx/OgsFxShaderGenerator.cpp

```diff
@@ -305,47 +305,31 @@ ShaderPtr OgsFxShaderGenerator::generate(const string& shaderName, ElementPtr el
     if (!vsPrivateUniforms.empty())
     {
         shader.addComment("Vertex stage uniform block: " + vsPrivateUniforms.name);
-        for (const Shader::Variable* uniform : vsPrivateUniforms.variableOrder)
-        {
-            emitUniform(*uniform, shader);
-        }
-        shader.newLine();
+        emitVariableBlock(vsPrivateUniforms, _syntax->getUniformQualifier(), shader, GlslSyntax::BLOCK_VARIABLE_SEPARATOR);
```

Comment:
There is a potential unexpected change in code path here, as the emitUniform method is virtual and can be overridden by ShaderGenerators to do different things. However I don't think that is being used currently in out derived generators so for now this is safe.

But maybe we should clean this up, and remove the emitUniform / emitConstant methods completely now, in favor of always using the more general emitVariable and emitVariableBlock? As their original usefulness has diminished now. And instead promote emitVariable and emitVariableBlock to public methods.